common/recommender/Baseline.js

```diff
@@ -15,27 +15,134 @@ class Baseline {
   constructor(history, options = {}) {
     this.domainCounts = history.reduce(this.countDomainOccurrences, new Map());
     this.options = options;
+    // Features that are extracted from URLs and need normalization.
+    // Key 0 holds the min, key 1 holds max, using arrays for brevity.
+    this.normalizeFeatures = {
+      idf: [1, 0],
```

Comment:
I think `idf: {min: 1, max: 0}` and normalizeFeatures[key].min`is a lot more readable, I would just go with an object